src/redux/books/bookSlice.js

```diff
@@ -0,0 +1,29 @@
+const ADD_BOOK = 'bookstore/books/addBook';
+const REMOVE_BOOK = 'bookstore/books/removeBook';
+
+const initialState = {
+  books: [],
+};
+
+// Add Reducers
+export default function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD_BOOK:
+      return { ...state };
```

Comment:
- [ ] Kindly make sure once the `ADD_BOOK` action is dispatched you must return the new state which contains the newly added book object and at this moment you are only returning the current state.

   - Suggestion:
       ` return state.books = [ ...state.books, action.payload]`